server/block/vine.go

```diff
@@ -0,0 +1,278 @@
+package block
+
+import (
+	"github.com/df-mc/dragonfly/server/block/cube"
+	"github.com/df-mc/dragonfly/server/block/model"
+	"github.com/df-mc/dragonfly/server/item"
+	"github.com/df-mc/dragonfly/server/world"
+	"github.com/go-gl/mathgl/mgl64"
+	"math/rand"
+)
+
+// Vines are climbable non-solid vegetation blocks that grow on walls.
+type Vines struct {
+	replaceable
+	transparent
+	empty
+
+	// NorthDirection is true if the vines are attached towards north.
+	NorthDirection bool
+	// EastDirection is true if the vines are attached towards east.
+	EastDirection bool
+	// SouthDirection is true if the vines are attached towards south.
+	SouthDirection bool
+	// WestDirection is true if the vines are attached towards west.
+	WestDirection bool
+}
+
+// FlammabilityInfo ...
+func (Vines) FlammabilityInfo() FlammabilityInfo {
+	return newFlammabilityInfo(15, 100, true)
+}
+
+// BreakInfo ...
+func (v Vines) BreakInfo() BreakInfo {
+	return newBreakInfo(0.2, alwaysHarvestable, shearsEffective, oneOf(v))
```

Comment:
Not always harvestable, requires shears or a silk touch axe

Reply:
Any ideas on how I should check for silk touch axes?

Reply:
Updated.